LEGO1/lego/sources/misc/legostorage.h

```diff
@@ -72,6 +73,37 @@ class LegoFile : public LegoStorage {
 	LegoResult SetPosition(LegoU32 p_position) override;
 	LegoResult Open(const char* p_name, LegoU32 p_mode);
 
+	// FUNCTION: LEGO1 0x10034430
+	LegoStorage* ReadVector3(Vector3* p_vec3)
+	{
+		Read(&(*p_vec3)[0], sizeof(float));
+		Read(&(*p_vec3)[1], sizeof(float));
+		Read(&(*p_vec3)[2], sizeof(float));
+		return this;
+	}
+
+	// FUNCTION: LEGO1 0x100343d0
+	LegoStorage* WriteVector3(Vector3 p_vec3)
```

Comment:
This takes a `Mx3DPointFloat` as a parameter. `100343a0` is the constructor called in `Act1State::NamedPlane::Serialize` before it is passed to `WriteVector3`. In the constructor you can tell by the virtual table address that is being assigned among other things. The generated constructor should also be annotated (see the bottom of `act1state.h` for an example)

Reply:
Yup, you're right.